server/src/internalClusterTest/java/org/opensearch/cluster/metadata/MetadataIndexStateServiceIT.java

```diff
@@ -53,20 +53,24 @@ public void testIndexCloseAndOpen() throws Exception {
                 .get();
         }
 
-        assertAcked(client().admin().indices().prepareClose(TEST_INDEX).get());
-
-        ClusterStateResponse clusterStateResponse = client().admin().cluster().prepareState().get();
-        IndexMetadata indexMetadata = clusterStateResponse.getState().metadata().index(TEST_INDEX);
-        assertEquals(IndexMetadata.State.CLOSE, indexMetadata.getState());
+        CloseIndexResponse closeIndexResponse = client().admin().indices().prepareClose(TEST_INDEX).get();
+        assertBusy(() -> {
+            assertTrue("Close operation should be acknowledged", closeIndexResponse.isAcknowledged());
+            assertEquals(
+                IndexMetadata.State.CLOSE,
+                client().admin().cluster().prepareState().get().getState().metadata().index(TEST_INDEX).getState()
+            );
+        }, 30, TimeUnit.SECONDS);
 
         OpenIndexResponse openIndexResponse = client().admin().indices().prepareOpen(TEST_INDEX).get();
-
-        assertTrue("Open operation should be acknowledged", openIndexResponse.isAcknowledged());
-        assertTrue("Open operation shards should be acknowledged", openIndexResponse.isShardsAcknowledged());
-
-        clusterStateResponse = client().admin().cluster().prepareState().get();
-        indexMetadata = clusterStateResponse.getState().metadata().index(TEST_INDEX);
-        assertEquals(IndexMetadata.State.OPEN, indexMetadata.getState());
+        assertBusy(() -> {
+            assertTrue("Open operation should be acknowledged", openIndexResponse.isAcknowledged());
+            assertTrue("Open operation shards should be acknowledged", openIndexResponse.isShardsAcknowledged());
+            assertEquals(
+                IndexMetadata.State.OPEN,
```

Comment:
I think ensureGreen here would accomplish the same thing?

Reply:
Thanks I have updated to use `ensureGreen`.

Reply:
So the for `_close` index we assert the state for `CLOSE` and for _open we use `ensureGreen`.